### PR TITLE
Cleanups for compatibility with future Bazel releases

### PR DESCRIPTION
**Pull request by laurentlb**
- Update dependencies
- Stop using Skylib lib.bzl (use more specific files)
- Avoid dictionary concatenation

Progress towards #489

### CODE REVIEW

File: haskell/nix.bzl [hunk 1]
```diff
@@ -192,9 +191,9 @@ def haskell_nixpkgs_packageset(name, base_attribute_path, repositories = {}, **k
       `@hackage//:{your-package-name}`
     """
 
-    repositories = {
-        "bazel_haskell_wrapper": "@io_tweag_rules_haskell//haskell:nix/default.nix",
-    } + repositories
+    repositories = dict(repositories)
+    repositories.setdefault("bazel_haskell_wrapper", "@io_tweag_rules_haskell//haskell:nix/default.nix")
```
Comment (guibou) [inline haskell/nix.bzl]: I had never thought about this usage of `setdefault` and I find it rather difficult to read. Why not:

```
{ "bazel_haskell_wrapper": "@io_tweag_rules_haskell//haskell:nix/default.nix"}.update(repositories)
￼```

Comment (laurentlb) [reply haskell/nix.bzl]: `dict.update` doesn't return any value.
I've updated the code to use the skylib function `dicts.add`

**Comment by laurentlb** [comment]
The failure on Linux is:
```
error cloning protobuf:
+ cd /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external
+ rm -rf /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/protobuf /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/protobuf
+ git clone https://github.com/protocolbuffers/protobuf.git /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/protobuf
bash: line 5: git: command not found
```

Is it an option to install `git` on the CI?

**Comment by Profpatsch** [comment]
> Is it an option to install `git` on the CI?

Sure, let me push the change.

**Comment by Profpatsch** [comment]
This should do the trick.

**Comment by laurentlb** [comment]
Thanks, tests pass now!

File: haskell/nix.bzl (hunk 1 above)
Comment (guibou) [reply haskell/nix.bzl]: Indeed about `dict.update`, I wrote this too quickly ;) Good solution with `dicts.add`.

**Pull request merged**